src/functions.jl

```diff
@@ -514,25 +541,48 @@ for T in (:Taylor1, :TaylorN)
         @inline function asin!(c::$T{T}, a::$T{T}, r::$T{T}, k::Int) where {T<:Number}
             if k == 0
                 a0 = constant_term(a)
-                @inbounds c[0] = asin( a0 )
-                @inbounds r[0] = sqrt( 1 - a0^2 )
+                if $T == Taylor1
+                    @inbounds c[0] = asin( a0 )
+                    @inbounds r[0] = sqrt( 1 - a0^2 )
+                else
+                    @inbounds c[0][1] = asin( a0 )
+                    @inbounds r[0][1] = sqrt( 1 - a0^2 )
+                end
                 return nothing
             end
             zero!(c, k)
-            if $T == Taylor1
-                @inbounds c[k] = (k-1) * r[1] * c[k-1]
-            else
-                @inbounds mul!(c[k], (k-1) * r[1], c[k-1])
-            end
-            @inbounds for i in 2:k-1
+            @inbounds for i in 1:k-1
                 if $T == Taylor1
                     c[k] += (k-i) * r[i] * c[k-i]
                 else
-                    mul!(c[k], (k-i) * r[i], c[k-i])
+                    mul!(c[k], r[i], c[k-i], scalar=k-i)
+                end
+            end
```

Comment:
Same comment about order of `if` and `for`

Reply:
Done 👍